addons/im_livechat/models/chatbot_script.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models, fields
+
+
+class ChatbotScript(models.Model):
+    _name = 'im_livechat.chatbot.script'
+    _description = 'Chatbot'
+    _inherit = ['image.mixin']
+    _inherits = {'utm.source': 'utm_source_id'}
+
+    active = fields.Boolean(default=True)
+    step_ids = fields.One2many('im_livechat.chatbot.script_step', 'chatbot_id',
+        copy=True, string='Script Steps')
+    utm_source_id = fields.Many2one('utm.source', string='UTM Source', ondelete='restrict', required=True)
+    operator_partner_id = fields.Many2one('res.partner', string='Bot Operator',
+        ondelete='restrict', required=True, copy=False)
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        """ Correctly copy the 'triggering_answer_ids' field from the original step_ids to the clone.
+        This needs to be done in post-processing to make sure we get references to the newly created
+        answers from the copy instead of references to the answers of the original.
+
+        This implementation assumes that the order of created steps and answers will be kept between
+        the original and the clone, using 'zip()' to match the records between the two. """
+
+        default = default or {}
+        default['name'] = self.name + _(' (copy)')
+
+        clone_chatbot_script = super().copy(default=default)
+
+        answers_map = {
+            original_answer: clone_answer
+            for clone_answer, original_answer
+            in zip(clone_chatbot_script.step_ids.answer_ids, self.step_ids.answer_ids)
+        }
+
+        for clone_step, original_step in zip(clone_chatbot_script.step_ids, self.step_ids):
```

Comment:
I used the same trick as PYDU for the survey optional questions thingy.
I'm not sure how this could be solved otherwise, do you have any suggestions?